code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,233 @@
+		to_chat(human_user, "<span class='warning'>You should be behind [src] to man it!</span>")
+		return TRUE
+	if(operator) //If there is already a operator then they're manning it.
+		if(!operator.interactee)
+			stack_trace("/obj/machinery/deployable/mounted/interact(mob/user) called by user [human_user] with an operator with a null interactee: [operator].")
+			operator = null //this shouldn't happen, but just in case
+		to_chat(human_user, "<span class='warning'>Someone's already controlling it.</span>")
+		return TRUE
+	if(human_user.interactee) //Make sure we're not manning two guns at once, tentacle arms.
+		to_chat(human_user, "<span class='warning'>You're already busy!</span>")
+		return TRUE
+	if(issynth(human_user) && !CONFIG_GET(flag/allow_synthetic_gun_use))
+		to_chat(human_user, "<span class='warning'>Your programming restricts operating heavy weaponry.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span class='notice'>[human_user] mans the [src]!</span>",
+		"<span class='notice'>You man the gun!</span>")
+
+	return ..()
+
+
+/obj/machinery/deployable/mounted/attackby(obj/item/I, mob/user, params) //This handles reloading the gun, if its in acid cant touch it.
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return
+
+	reload(user, I)
+
+
+///Reloads gun
+/obj/machinery/deployable/mounted/proc/reload(mob/user, ammo_magazine)
+	if(!istype(ammo_magazine, /obj/item/ammo_magazine))
+		return
+
+	var/obj/item/ammo_magazine/ammo = ammo_magazine
+	var/obj/item/weapon/gun/gun = internal_item
+	if(!istype(gun, ammo.gun_type))
+		return
+	if(ammo.current_rounds <= 0)
+		to_chat(user, "<span class='warning'>[ammo] is empty!</span>")
+		return
+
+	if(gun.current_mag)
+		gun.unload(user,0,1)
+		update_icon_state()
+
+	var/tac_reload_time = max(0.5 SECONDS, 1.5 SECONDS - user.skills.getRating("firearms") * 5)
+	if(!do_after(user, tac_reload_time, TRUE, src, BUSY_ICON_FRIENDLY))
+		return
+	
+	gun.reload(user, ammo_magazine)
+	update_icon_state()
+
+///Sets the user as manning the internal gun
+/obj/machinery/deployable/mounted/on_set_interaction(mob/user)
+	operator = user
+
+	. = ..()
+
+	var/obj/item/weapon/gun/gun = internal_item
+
+	if(!gun)
+		CRASH("[src] has been deployed and attempted interaction with [operator] without having a gun. This shouldn't happen.")
+	
+	
+
+	RegisterSignal(operator, COMSIG_MOB_MOUSEDOWN, .proc/start_fire)
+	gun.RegisterSignal(operator, COMSIG_MOB_MOUSEUP, /obj/item/weapon/gun/proc/stop_fire)
+	RegisterSignal(operator, COMSIG_MOB_MOUSEDRAG, .proc/change_target)
+
+	for(var/datum/action/action AS in gun.actions)
+		action.give_action(operator)
+	var/obj/screen/ammo/hud = operator.hud_used.ammo
+	hud.add_hud(operator, internal_item)
+	hud.update_hud(operator, internal_item)
+
+	gun.gun_user = operator
```

Comment:
If my gun_user hard del protection PR is merged before yours, this will have to use set_gun_user. Else, I'll do it myself, but please add a todo in comment